corehq/apps/app_manager/suite_xml/post_process/instances.py

```diff
@@ -366,7 +372,9 @@ def selected_cases_instances(app, instance_name):
 
 @register_factory('results')
 def remote_instances(app, instance_name):
-    return Instance(id=instance_name, src=f'jr://instance/remote/{instance_name}')
+    if is_valid_results_instance_name(app, instance_name):
+        return Instance(id=instance_name, src=f'jr://instance/remote/{instance_name}')
+    return None
```

Comment:
Haven't read over the whole PR yet, so this might be in there somewhere, but as a reminder to myself, what happens when this returns `None`?  Should it instead raise some exception (whether a 400 or 500)?

Reply:
None is returned [here](https://github.com/dimagi/commcare-hq/blob/3771ff8f14c4ae6d3db8f2451cc83c634c956f45/corehq/apps/app_manager/suite_xml/post_process/instances.py#L425) and the instance name is returned in `unknown_instance_ids` when it's called by[add_missing_instances](https://github.com/dimagi/commcare-hq/blob/b719aeaac9243114e56d0ea4d02106ed1ca08555/corehq/apps/app_manager/xform.py#L869). This ends up raising [XFormValidationError](https://github.com/dimagi/commcare-hq/blob/b719aeaac9243114e56d0ea4d02106ed1ca08555/corehq/apps/app_manager/xform.py#L879)

Reply:
I considered raising the `UnknownInstanceError` but decided to handle it the same way as `null factory` https://github.com/dimagi/commcare-hq/blob/3771ff8f14c4ae6d3db8f2451cc83c634c956f45/corehq/apps/app_manager/suite_xml/post_process/instances.py#L317

Reply:
cool thanks - I do now see that the reports instance factory similarly returns `None` in some cases.